src/LinuxTracing/TracerImpl.cpp

```diff
@@ -566,15 +587,34 @@ void TracerImpl::InitSwitchesStatesNamesVisitor() {
   event_processor_.AddVisitor(switches_states_names_visitor_.get());
 }
 
-bool TracerImpl::OpenContextSwitchAndThreadStateTracepoints(const std::vector<int32_t>& cpus) {
+bool TracerImpl::OpenContextSwitchAndThreadStateTracepoints(
+    const std::vector<int32_t>& cpus,
+    const CaptureOptions::ThreadStateChangeCallStackCollection
+        thread_state_change_callstack_collection,
+    const CaptureOptions::UnwindingMethod unwinding_method) {
   ORBIT_SCOPE_FUNCTION;
   std::vector<TracepointToOpen> tracepoints_to_open;
+  absl::flat_hash_set<uint64_t>* current_sched_switch_ids = &sched_switch_ids_;
+  absl::flat_hash_set<uint64_t>* current_sched_wakeup_ids = &sched_wakeup_ids_;
+  if (thread_state_change_callstack_collection ==
+          CaptureOptions::kThreadStateChangeCallStackCollection &&
+      thread_state_change_callstack_method_ == CaptureOptions::kDwarf) {
+    current_sched_switch_ids = &sched_switch_with_stack_ids_;
+    current_sched_wakeup_ids = &sched_wakeup_with_stack_ids_;
+  }
+  if (thread_state_change_callstack_collection ==
```

Comment:
```suggestion
 else if (thread_state_change_callstack_collection ==
```

Reply:
also consider nesting the ifs and have an "if (thread_state_.... ==  ...CallstackCollection) then" surrounding.

Reply:
Done.